File: src/main/java/com/rex/student/Student.java
package com.rex.student;

public class Student {
    String id;
    String name;
    int english;
    int math;

    public Student(String name, int english, int math) {
        this.name = name;
        this.english = english;
        this.math = math;
    }

    public int highest () {
//        int max = (english > math) ? english : math;

//        if (english > math) {
//            max = english;
//        } else {
//            max = math;
//        }

        return (english > math) ? english : math;
    }

    public void print() {
        System.out.print(name + "\t\t" + english + "\t" + math +
                "\t" + getAverage() + "\t" +
                ((getAverage() >= 60) ? "PASS" : "FAILED"));
//        if (getAverage() >= 60) {
//            System.out.println("\tPASS");
//        } else {
//            System.out.println("\tFAILED");
//        }

        int average = getAverage();
        char grading = 'F';

        switch (average / 10) {
            case 10:
            case 9:
                grading = 'A';
                break;
            case 8:
                grading = 'B';
                break;
            case 7:
                grading = 'C';
                break;
            case 6:
                grading = 'D';
                break;
            default:
                grading = 'F';
        }
        /*if (average >= 90 && average <= 100) {
            grading = 'A';
        } else if (average >= 80 && average <= 89) {
            grading = 'B';
        } else if (average >= 70 && average <= 79) {
            grading = 'C';
        } else if (average >= 60 && average <= 69) {
            grading = 'D';
        }*/
        System.out.println("\t" + grading);
    }

    public int getAverage() {
        return (english + math) / 2;
    }
}

File: src/main/kotlin/com/rex/student/student.kt
package com.rex.student

import java.util.*

fun main(args: Array<String>) {
//    userInput()
    val stu = Human("rex", 66, 88)
    stu.print()
    println("High score: ${stu.highest()}")
}

class Human(var name :String?,var english :Int, var math :Int){
    fun print() {
        println("$name\t$english\t$math\t${getAverage()}\t${passOrFailed()}\t${grading()}")
//        print (name + "\t" + english + "\t" + math +
//                "\t" + getAverage() + "\t" +
//                )
//        println("\t" + grading())
    }

    fun grading() = when(getAverage()) {
        in 90..100 -> 'A'
        in 80..89 -> 'B'
        in 70..79 -> 'C'
        in 60..69 -> 'D'
        else -> 'F'
    }

    fun passOrFailed() = if(getAverage() >= 60) "PASS" else "FAILED"

/*
    fun grading() :Char{
        var grading = when(getAverage()) {
            in 90..100 -> 'A'
            in 80..89 -> 'B'
            in 70..79 -> 'C'
            in 60..69 -> 'D'
            else -> 'F'
        }
        return grading
    }
*/

    fun getAverage() = (english + math) / 2

    fun highest() = if (english > math) {
        println("english")
        english
    } else {
        println("math")
        math
    }

    fun nameCheck() = name?.length
}

private fun userInput() {
    val scanner = Scanner(System.`in`)
//    println("Please enter student's name: ")
    val name = null
    println("Please enter student's english: ")
    val english = scanner.nextInt()
    println("Please enter student's math: ")
    val math = scanner.nextInt()
    val stu = Human(name, english, math)
    stu.print()
    stu.nameCheck()
}
